GPU/Common/TextureCacheCommon.cpp

```diff
@@ -1894,7 +1894,7 @@ void TextureCacheCommon::ApplyTextureFramebuffer(VirtualFramebuffer *framebuffer
 
 			// Since we started/ended render passes, might need these.
 			gstate_c.Dirty(DIRTY_DEPAL);
-			gstate_c.SetUseShaderDepal(true);
+			gstate_c.SetUseShaderDepal(true, gstate.getClutIndexStartPos() == 0 && gstate.getClutIndexMask() <= clutTexture.rampLength);
```

Comment:
Nice.  I think we need to check a couple other things, though:
 * Is it 4444/8888?  Drop (although maybe could add to shader.)
 * Is the shift other than 5/10/11/whatever aligns with the format?  Mask?  If not aligned, drop.

We would have to adjust the shader to allow variable masks.

-[Unknown]

 force off for 4444/8888 or else handle 4444 in the shader, though.

Reply:
Yeah, didn't get to that yesterday, but now I have :)